### ISSUE TITLE
Find the best parameters for summarization of papers

### ISSUE THREAD

**Issue by InAnYan** [InAnYan/jabref#36]

**Comment by InAnYan** [comment]
https://github.com/GoogleCloudPlatform/generative-ai/blob/main/language/use-cases/document-summarization/summarization_large_documents_langchain.ipynb

**Comment by ThiloteE** [comment]
1) Stuffing: just put the whole contents of document into user message
2) Map-reduce: summarize small chunks of text and combine these summarizations into one
3) Refine: (not really sure about this one), summarize a small chunk of text, then give the AI this summarization + new small chunk and ask it to refine the current summarization, and do this recursively,
4) Only detect and embed the "Abstract" in the PDF

**Comment by ThiloteE** [comment]
I do not think this is a low priority issue. It is a normal to high priority issue, as summarization is the main goal of this GSoC project. Though, it is an issue that may not be important in this week. So, just use the milestone table to check, if you should give it attention this week or not.

**Comment by InAnYan** [comment]
Ah, I forgot, sorry, yes.

I used `low-priority` to filter out issues for Week 1, but then I realized here are `Milestones` on GitHub, so we don't need `low-priority`

**Comment by ThiloteE** [comment]
It is ok to differentiate between low, normal or high priority. The more issues there are, the more it is necessary to prioritize.

**Comment by koppor** [comment]
> 1. Stuffing: just put the whole contents of document into user message

[...]

Ruslan found one more: https://pashpashpash.substack.com/p/tackling-the-challenge-of-document:

> Break Down the Document into "Key Topics"

**Comment by koppor** [comment]
As long as there is some basic functionality working - not causing WTFs of users, I am open to move this to week 11 or week 12.

**Comment by ThiloteE** [comment]
Inspiration for summarization (system) prompts:

https://github.com/ahmetbersoz/chatgpt-prompts-for-academic-writing
https://aihabit.net/chatgpt-prompts-for-summaries/
https://nerdschalk.com/chatgpt-prompts-to-summarize-text/

Also, I have seen reports that the word "summarization" can actually lead to worse results than other commands without it. For example, "Render the input as a distilled list of succinct statements, assertions, associations, concepts, analogies and metaphors. The idea is to capture as much conceptually as possible but with as few words as possible." or "You are a journalism editor who must condense this large body of text to a more concise  presentation for a reader with only a few minutes to read and must be given the essential content." might lead to similar or better results.

**Comment by ThiloteE** [comment]
Ultimately, it depends on the model that is used to do the summarization, if any particular prompt works well or not. So, if we have basic functionality, it's fine.

For anything advanced, users will have to use the "chatting with AI" tab or we have to expose this prompt to be editable in the preferences.
